modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -476,5 +476,39 @@ namespace {{packageName}}.Client
             return buffer;
         }
         {{/supportsUWP}}
+
+        /// <summary>
+        /// Convert params to key/value pairs. 
+        /// Use collectionFormat to properly format lists and collections.
+        /// </summary>
+        /// <param name="name">Key name.</param>
+        /// <param name="value">Value object.</param>
+        /// <returns>A list of KeyValuePairs</returns>
+        public IEnumerable<KeyValuePair<string, string>> ParameterToKeyValuePairs(string collectionFormat, string name, object value)
+        {
+            var parameters = new List<KeyValuePair<string, string>>();
+
+            if (IsCollection(value) && collectionFormat == "multi")
```

Comment:
We can also add support for other collectionFormats as well  (e.g pipe, slash etc).